nixos/modules/services/web-servers/minio.nix

```diff
@@ -34,7 +34,7 @@ in
       description = lib.mdDoc "The list of data directories or nodes for storing the objects. Use one path for regular operation and the minimum of 4 endpoints for Erasure Code mode.";
     };
 
-    configDir = mkOption {
+    certsDir = mkOption {
```

Comment:
[I am new touchig module, and felt this name should be `certsDir` instead of `configDir` but now after this change, it feel that it might break anyone who would be overriding this.
Looking for guidance on:
-> Should we keep it as `configDir` ?
-> Is there a way for us to see if any package breaks with this change ?

Thanks.

Reply:
@mfrw I thik you can rename the option and use [`mkRenamedOptionModuleWith`](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20mkRenamedOptionModuleWith&type=code) to keep the old name but notify the user about the rename.

Reply:
@mfrw Although I am no expert, but maybe this would do the trick:

```diff
diff --git a/nixos/modules/services/web-servers/minio.nix b/nixos/modules/services/web-servers/minio.nix
index 4ddd90bfa3ed..8a3b482d98a2 100644
--- a/nixos/modules/services/web-servers/minio.nix
+++ b/nixos/modules/services/web-servers/minio.nix
@@ -89,6 +89,11 @@ in
   };
 
   config = mkIf cfg.enable {
+    imports = [(lib.mkRenamedOptionModuleWith { 
+      sinceRelease = 2411;
+      from = ["services" "minio" "configDir"]; 
+      to = ["services" "minio" "certsDir" ];
+    })];
     warnings = optional ((cfg.accessKey != "") || (cfg.secretKey != "")) "services.minio.`accessKey` and services.minio.`secretKey` are deprecated, please use services.minio.`rootCredentialsFile` instead.";
 
     systemd = lib.mkMerge [{

```

Reply:
thank you @jrester for the tip.